src/app/drawer/drawer.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, signal } from '@angular/core';
+import { MockDataService } from '../mockData/Table data';
+import { TableDataInterface } from '../shared/interfaces';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DrawerService {
+  isDrawerOpen = signal(false);
+  fullData!: TableDataInterface[];
+  tableData = signal(this.fullData);
+
+  constructor(public mockDataService: MockDataService) {
+    this.fullData = this.mockDataService.getData();
+  }
+
+  getTableData() {
```

Comment:
if you would like you can use rxjs to simulate a backend response: return of(this.fullData)
this way in the component we'll get an observable you can subscribe to and the scenario is more similar to a BE call